###### Running code on Seperate Notebook for Lusitania Data 

In [11]:
import pandas as pd 
import numpy as np
import math
Lusitania = pd.read_csv('lusitania.csv')

In [12]:
Lusitania

Family name       Title       Personal name   Fate   Age  \
0        CAMERON         Mr.          Charles W.   Lost    38   
1     CARR-JONES         Mr.                  E.   Lost    37   
2      DRAKEFORD         Mr.              Edward  Saved    30   
3        HAWKINS         Mr.              Handel  Saved    25   
4      HEMINGWAY         Mr.        John William  Saved    27   
5       ANDERSON         Mr.        James Clarke   Lost    48   
6       ANDERSON         Mr.                John   Lost   NaN   
7         BATTLE         Mr.               James  Saved   NaN   
8         BESTIC         Mr.       Albert Arthur  Saved    24   
9         BOWDEN         Mr.             Joseph   Saved  19 ?   
10         BROWN         Mr.  John Manson George   Lost    57   
11        CANNON         Mr.        Robert Henry   Lost    50   
12       CLINTON         Mr.              George  Saved   NaN   
13         CLYDE         Mr.                Neil  Saved   NaN   
14         CRANK         Mr.                John   Lost   NaN   
15    CUNNINGHAM         Mr.              Edward  Saved   NaN   
16        DAVIES         Mr.                John  Saved   NaN   
17          EGAN         Mr.             William  Saved   NaN   
18        ELLIOT         Mr.       Arthur Graham   Lost   NaN   
19        ELLIOT         Mr.                Fred  Saved   NaN   
20         EVANS         Mr.      Thomas William   Lost   NaN   
21         FAYLE         Mr.              Edward  Saved   NaN   
22      FLEMMING         Mr.             Herbert   Lost   NaN   
23         GARRY         Dr.              Joseph   Lost    25   
24          GUNN         Mr.                  C.  Saved   NaN   
25        HARMAN         Mr.                John  Saved    56   
26       HEFFORD         Mr.               Percy   Lost   NaN   
27      HEIGHWAY         Mr.         Edward John  Saved    41   
28      HENNESEY         Mr.               Frank  Saved   NaN   
29        HERMAN         Mr.              Albert   Lost   NaN   
...          ...         ...                 ...    ...   ...   
1931      THOMAS        Mr.     William Stanford   Lost    44   
1932      THOMAS        Mr.         George Henry   Lost   NaN   
1933      THOMAS        Mr.                 John  Saved   NaN   
1934       THORN        Mr.       Alfred Richard   Lost   NaN   
1935    THORNLEY        Mr.               George  Saved   NaN   
1936       ULGAR        Mr.                 Jean   Lost   NaN   
1937   Van HOECK        Mr.           Julius E.   Saved    27   
1938      VARNEY        Mr.                 John   Lost   NaN   
1939      VERDIN        Mr.              Francis   Lost   NaN   
1940     WALLACE        Mr.                James  Saved   NaN   
1941       WEIGH        Mr.              Edward�  Saved   NaN   
1942        WEIR        Mrs.            Margaret   Lost    51   
1943     WHALLEY        Mr.               Leslie  Saved   NaN   
1944     WIGGINS        Mr.              William  Saved    23   
1945   WILKINSON        Mr.            Mr. James   Lost   NaN   
1946    WILLIAMS        Mr.                 John   Lost   NaN   
1947    WILLIAMS        Mr.           Mr. Harold   Lost   NaN   
1948    WILLIAMS        Mr.               Robert   Lost   NaN   
1949    WILLIAMS        Mr.       Robert Neptune   Lost   NaN   
1950    WILLIAMS        Mr.           William H.  Saved   NaN   
1951      WILMIN        Mr.              William  Saved   NaN   
1952      WILSON        Mr.               Robert   Lost   NaN   
1953     WISEMAN        Mr.                James   Lost   NaN   
1954        WOOD        Mr.               Walter  Saved    31   
1955        WOOD        Mr.         Henry Edward   Lost    35   
1956        WOOD        Mr.               Alfred  Saved   NaN   
1957        WOOD        Mr.        Wallace Edkin   Lost   NaN   
1958      WRIGHT  Master (?)       Harold Joseph   Lost   NaN   
1959       WYNNE      Master              George  Saved    16   
1960       WYNNE        Mr.       Joseph

###### Interesting enough the max and min age of the Lusitania data set is Infant or inf.  This just means there are a few strings present in the data set I need to clean out before looking at numerical averages.  

In [13]:
Lusitania['Age'].max()

'Infant'

In [14]:
Lusitania['Age'].min()

inf

In [15]:
Lusitania.describe()

Value
count  1961.000000
mean      0.999490
std       0.022582
min       0.000000
25%       1.000000
50%       1.000000
75%       1.000000
max       1.000000

###### Usings the pandas dataframe I am creating a pivot table based on Passenger/Crew, and Department so I can better understand the rate of survival betwee the different subsets of passengers.  

In [16]:
Lusitania[['Passenger/Crew', 'Department/Class', 'Fate']].pivot_table

L1 = Lusitania.pivot_table('Title', index=['Passenger/Crew', 'Department/Class'], columns=['Fate'], fill_value=0, aggfunc=len, margins=True)
L1.columns


#Calculating the Survival rate as Saved over total number of passengers or represented here as 
#total of (Lost, Saved, Not on Board, Saved (but died from trauma))
L1['Survival Rate'] = L1['Saved'] / (L1['Lost'] + L1['Saved'] + L1['Not on board'])
L1

Fate                                                Lost  Not on board  Saved  \
Passenger/Crew Department/Class                                                 
Crew           Band                                  2.0           0.0    3.0   
               Deck                                 32.0           0.0   37.0   
               Engineering                         201.0           0.0  112.0   
               Victualling                         167.0           0.0  138.0   
Passenger      Saloon                              177.0           1.0  112.0   
               Second                              372.0           0.0  227.0   
               Third                               236.0           0.0  134.0   
               Third (Distressed British Seaman)     3.0           0.0    0.0   
Stowaway       Stowaway                              3.0           0.0    0.0   
All                                               1193.0           1.0  763.0   

Fate                                              Saved (died from trauma)  \
Passenger/Crew Department/Class                                              
Crew           Band                                                    0.0   
               Deck                                                    0.0   
               Engineering                                             0.0   
               Victualling                                             1.0   
Passenger      Saloon                                                  1.0   
               Second                                                  2.0   
               Third                                                   0.0   
               Third (Distressed British Seaman)                       0.0   
Stowaway       Stowaway                                                0.0   
All                                                                    4.0   

Fate                                                 All  Survival Rate  
Passenger/Crew Department/Class                                          
Crew           Band                                  5.0       0.600000  
               Deck                                 69.0       0.536232  
               Engineering                         313.0       0.357827  
               Victualling                         306.0       0.452459  
Passenger      Saloon                              291.0       0.386207  
               Second                              601.0       0.378965  
               Third                               370.0       0.362162  
               Third (Distressed British Seaman)     3.0       0.000000  
Stowaway       Stowaway                              3.0       0.000000  
All                                               1961.0       0.389882

###### Above we can see the Band had the highest survival rate followed by the member fo the Deck.  Overall, the members of the Crew had a higher survial rate than the passengers . 

###### A pivot table to how the rate of survival was contained among the various titles of the Lusitania 

In [17]:
Lusitania.pivot_table('Age', index=['Title', 'Fate'], columns=['Passenger/Crew'], aggfunc=len)

Passenger/Crew                                 Crew  Passenger  Stowaway
Title               Fate                                                
Captain             Lost                        NaN        2.0       NaN
                    Saved                       1.0        NaN       NaN
Commander           Lost                        NaN        1.0       NaN
Detective-Inspector Saved                       NaN        1.0       NaN
Dr.                 Lost                        2.0        2.0       NaN
                    Saved                       NaN        7.0       NaN
Father              Lost                        NaN        1.0       NaN
Lady                Saved                       NaN        2.0       NaN
Lieutenant          Lost                        NaN        1.0       NaN
                    Saved                       NaN        1.0       NaN
Master              Lost                        3.0       57.0       NaN
                    Saved                       6.0       29.0       NaN
                    Saved (died from trauma)    NaN        1.0       NaN
Master (?)          Lost                        4.0        NaN       NaN
Miss                Lost                       11.0      143.0       NaN
                    Saved                       4.0       80.0       NaN
Mr.                 Lost                      231.0      410.0       NaN
                    Saved                     151.0      245.0       NaN
                    Saved (died from trauma)    NaN        1.0       NaN
Mr.                 Lost                      146.0        NaN       3.0
                    Saved                     123.0        NaN       NaN
                    Saved (died from trauma)    1.0        NaN       NaN
Mr. (later Sir)     Saved                       NaN        1.0       NaN
Mrs.                Lost                        5.0      165.0       NaN
                    Not on board                NaN        1.0       NaN
                    Saved                       5.0      100.0       NaN
                    Saved (died from trauma)    NaN        1.0       NaN
Professor           Lost                        NaN        1.0       NaN
                    Saved                       NaN        2.0       NaN
Reverend            Lost                        NaN        3.0       NaN
                    Saved                       NaN        4.0       NaN
Sir                 Lost                        NaN        1.0       NaN
Sister              Lost                        NaN        1.0       NaN
Surgeon-Major       Saved                       NaN        1.0       NaN

###### In my pivot table above I have a few weird titles Mr. (later Sir), and Master (?).  I am going to remove them, creating a new data set where Mr. (Later Sir) and Master (?) will be joing with Mr. and Master respectively.  

In [24]:
new_Title = []

for i in Lusitania['Title']:
    x = i.split(' ')
    w = x[0].split(',')    
    y = w[0].strip()
    new_Title.append(y)

In [25]:
Lusitania['Title'] = new_Title

In [26]:
L2 = Lusitania.pivot_table('Family name', index=['Title'], columns=['Passenger/Crew', 'Fate'], fill_value=0, aggfunc=len, margins=True)

In [27]:
L2

Passenger/Crew        Crew                                 Passenger  \
Fate                  Lost  Saved Saved (died from trauma)      Lost   
Title                                                                  
Captain                0.0    1.0                      0.0       2.0   
Commander              0.0    0.0                      0.0       1.0   
Detective-Inspector    0.0    0.0                      0.0       0.0   
Dr.                    2.0    0.0                      0.0       2.0   
Father                 0.0    0.0                      0.0       1.0   
Lady                   0.0    0.0                      0.0       0.0   
Lieutenant             0.0    0.0                      0.0       1.0   
Master                 7.0    6.0                      0.0      57.0   
Miss                  11.0    4.0                      0.0     143.0   
Mr.                  377.0  274.0                      1.0     410.0   
Mrs.                   5.0    5.0                      0.0     165.0   
Professor              0.0    0.0                      0.0       1.0   
Reverend               0.0    0.0                      0.0       3.0   
Sir                    0.0    0.0                      0.0       1.0   
Sister                 0.0    0.0                      0.0       1.0   
Surgeon-Major          0.0    0.0                      0.0       0.0   
All                  402.0  290.0                      1.0     788.0   

Passenger/Crew                                                   Stowaway  \
Fate                Not on board  Saved Saved (died from trauma)     Lost   
Title                                                                       
Captain                      0.0    0.0                      0.0      0.0   
Commander                    0.0    0.0                      0.0      0.0   
Detective-Inspector          0.0    1.0                      0.0      0.0   
Dr.                          0.0    7.0                      0.0      0.0   
Father                       0.0    0.0                      0.0      0.0   
Lady                         0.0    2.0                      0.0      0.0   
Lieutenant                   0.0    1.0                      0.0      0.0   
Master                       0.0   29.0                      1.0      0.0   
Miss                         0.0   80.0                      0.0      0.0   
Mr.                          0.0  246.0                      1.0      3.0   
Mrs.                         1.0  100.0                      1.0      0.0   
Professor                    0.0    2.0                      0.0      0.0   
Reverend                     0.0    4.0                      0.0      0.0   
Sir                          0.0    0.0                      0.0      0.0   
Sister                       0.0    0.0                      0.0      0.0   
Surgeon-Major                0.0    1.0                      0.0      0.0   
All                          1.0  473.0                      3.0      3.0   

Passenger/Crew          All  
Fate                         
Title                        
Captain                 3.0  
Commander               1.0  
Detective-Inspector     1.0  
Dr.                    11.0  
Father                  1.0  
Lady                    2.0  
Lieutenant              2.0  
Master                100.0  
Miss                  238.0  
Mr.                  1312.0  
Mrs.                  277.0  
Professor               3.0  
Reverend                7.0  
Sir                     1.0  
Sister                  1.0  
Surgeon-Major           1.0  
All                  1961.0

###### To create Ratios of Survival Based on Title I am splitting up the Data into 2 groups.  Survival Rate by Crew and Survial Rate by Passgenger

In [33]:
#Creating a data set of using just the crew information.

Crew = L2['Crew']

Crew['Survival Rate'] = Crew['Saved'] / (Crew['Saved'] + Crew['Lost'] + Crew['Saved (died from trauma)'])

/Users/punkymonkey/anaconda/lib/python2.7/site-packages/ipykernel/__main__.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  app.launch_new_instance()


In [36]:
print Crew

Fate                  Lost  Saved  Saved (died from trauma)  Survival Rate
Title                                                                     
Captain                0.0    1.0                       0.0       1.000000
Commander              0.0    0.0                       0.0            NaN
Detective-Inspector    0.0    0.0                       0.0            NaN
Dr.                    2.0    0.0                       0.0       0.000000
Father                 0.0    0.0                       0.0            NaN
Lady                   0.0    0.0                       0.0            NaN
Lieutenant             0.0    0.0                       0.0            NaN
Master                 7.0    6.0                       0.0       0.461538
Miss                  11.0    4.0                       0.0       0.266667
Mr.                  377.0  274.0                       1.0       0.420245
Mrs.                   5.0    5.0                       0.0       0.500000
Professor              0.

###### Above we can see that among the crew we can see of t he 402 crew members 290 were saved or 41.8%.  

In [37]:
#Creating a dataframe to explore the passengers survival rate
Passenger = L2['Passenger']

In [39]:
Passenger['Survival Rate'] = Passenger['Saved'] / (Passenger['Saved'] + Passenger['Lost'] + Passenger['Not on board'] + Passenger['Saved (died from trauma)'])

/Users/punkymonkey/anaconda/lib/python2.7/site-packages/ipykernel/__main__.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  if __name__ == '__main__':


In [42]:
Passenger

Fate                  Lost  Not on board  Saved  Saved (died from trauma)  \
Title                                                                       
Captain                2.0           0.0    0.0                       0.0   
Commander              1.0           0.0    0.0                       0.0   
Detective-Inspector    0.0           0.0    1.0                       0.0   
Dr.                    2.0           0.0    7.0                       0.0   
Father                 1.0           0.0    0.0                       0.0   
Lady                   0.0           0.0    2.0                       0.0   
Lieutenant             1.0           0.0    1.0                       0.0   
Master                57.0           0.0   29.0                       1.0   
Miss                 143.0           0.0   80.0                       0.0   
Mr.                  410.0           0.0  246.0                       1.0   
Mrs.                 165.0           1.0  100.0                       1.0   
Professor              1.0           0.0    2.0                       0.0   
Reverend               3.0           0.0    4.0                       0.0   
Sir                    1.0           0.0    0.0                       0.0   
Sister                 1.0           0.0    0.0                       0.0   
Surgeon-Major          0.0           0.0    1.0                       0.0   
All                  788.0           1.0  473.0                       3.0   

Fate                 Survival Rate  
Title                               
Captain                   0.000000  
Commander                 0.000000  
Detective-Inspector       1.000000  
Dr.                       0.777778  
Father                    0.000000  
Lady                      1.000000  
Lieutenant                0.500000  
Master                    0.333333  
Miss                      0.358744  
Mr.                       0.374429  
Mrs.                      0.374532  
Professor                 0.666667  
Reverend                  0.571429  
Sir                       0.000000  
Sister                    0.000000  
Surgeon-Major             1.000000  
All                       0.373913

###### Above we can see that of the 788 passengers that 473 or only 37.4% survived the Lusitania.  It appears you had much better odds being a crew member on board the Lusitania.  Although, given being a passenger there was a subgroup which had a higher survial rate, and that was Dr. 